### PR TITLE
feat(api): add `discovered_peers` to `/network/info`

### PR DESCRIPTION
**Pull request by youngjoon-lee**
## 1. What does this PR implement?

Monitoring how many peers have been discovered is useful. The existing `connected_peers` field doesn't tell us that, because libp2p doesn't actively establish connections to the discovered peers until it becomes necessary. 

## 2. Does the code have enough context to be clearly understood?

Yes

## 3. Who are the specification authors and who is accountable for this PR?

@youngjoon-lee 

## 4. Is the specification accurate and complete?

N/A

## 5. Does the implementation introduce changes in the specification?

N/A

## Checklist

> [!WARNING]  
> Do not merge the PR if any of the following is missing:

* [x] 1. The PR title follows the Conventional Commits [specification](https://www.notion.so/How-to-open-a-Pull-Request-215261aa09df80deb538ed59f5e4e0b5).
* [x] 2. Description added.
* [x] 3. Context and links to Specification document(s) added.
* [x] 4. Main contact(s) (developers and specification authors) added
* [x] 5. Implementation and Specification are 100% in sync including changes. This is critical.
* [x] 6. Link PR to a specific milestone.
* [x] 7. New or changed logs were reviewed against the [logging guidelines](https://github.com/logos-blockchain/logos-blockchain/blob/master/CONTRIBUTING.md#logging-guidelines).

### CODE REVIEW

File: services/network/src/backends/libp2p/swarm/mod.rs [hunk 1]
```diff
@@ -266,6 +266,13 @@ impl<R: Clone + Send + RngCore + 'static> SwarmHandler<R> {
                 self.connect(dial);
             }
             NetworkCommand::Info { reply } => {
+                let discovered_peers: Vec<PeerId> = self
+                    .swarm
+                    .kademlia_discovered_peers()
+                    .into_iter()
+                    .map(|peer_info| peer_info.peer_id)
+                    .collect();
+                let n_discovered_peers = discovered_peers.len();
```
Comment (hansieodendaal) [inline services/network/src/backends/libp2p/swarm/mod.rs:275]: Are we sure that the Kademlia helper used here will not the local node as well? I think `kademlia_discovered_peers()` returns every Kademlia entry that has an address.

We could possibly filter out `*swarm.local_peer_id()` from `kademlia_discovered_peers()` before building `discovered_peers` and `n_discovered_peers`.

**Review by hansieodendaal — COMMENTED** [review]
This will be a great addition.

Maybe we should add a targeted unit/integration test, for example a focused test covering an empty routing table and a routing table containing the local peer plus remote peers.

**Review by danielSanchezQ — APPROVED** [review]

File: services/network/src/backends/libp2p/swarm/mod.rs (hunk 1 above)
Comment (youngjoon-lee) [reply services/network/src/backends/libp2p/swarm/mod.rs:275]: The local_peer_id is excluded by libp2p kademlia: https://github.com/libp2p/rust-libp2p/blob/9698607c72ea7b8f5e661963ab0e24384aef1e4e/protocols/kad/src/kbucket.rs#L142-L146.

Since this PR just implements API, I don't want to change the implementation of `lb_libp2p::behaviour::kademlia::Swarm::kademlia_discovered_peers`. I want the API to show exactly what the other crates get and use from the `kademlia_discovered_peers` function.

Also, I didn't add a test for an empty set because it's so trivial.

**Review by hansieodendaal — APPROVED** [review]
LGTM

File: services/network/src/backends/libp2p/swarm/mod.rs (hunk 1 above)
Comment (youngjoon-lee) [reply services/network/src/backends/libp2p/swarm/mod.rs:275]: @hansieodendaal 
Just added tests c7b376e8d. Thanks!
